src/Microsoft.TemplateEngine.Cli/TemplatePackageCoordinator.cs

```diff
@@ -253,53 +275,163 @@ private async Task<CreationResultStatus> EnterUpdateFlowAsync(INewCommandInput c
+            cancellationToken.ThrowIfCancellationRequested();
+
             CreationResultStatus result = CreationResultStatus.Success;
-            if (commandInput.ToUninstallList.Count > 0 && commandInput.ToUninstallList[0] != null)
+            if (commandInput.ToUninstallList.Count <= 0 || commandInput.ToUninstallList[0] == null)
             {
-                var managedSourcedGroupedByProvider = await _engineEnvironmentSettings.SettingsLoader.TemplatesSourcesManager.GetManagedSourcesGroupedByProvider().ConfigureAwait(false);
-                bool sourceToUninstallFound = false;
-                foreach (var (provider, sources) in managedSourcedGroupedByProvider)
+                //display all installed template packages
+                await DisplayInstalledTemplatesSources(commandInput, cancellationToken).ConfigureAwait(false);
+                return result;
+            }
+
+            Dictionary<IManagedTemplatesSourcesProvider, List<IManagedTemplatesSource>> sourcesToUninstall;
+            (result, sourcesToUninstall) = await DetermineSourcesToUninstall(commandInput, cancellationToken).ConfigureAwait(false);
+
+            foreach (KeyValuePair<IManagedTemplatesSourcesProvider, List<IManagedTemplatesSource>> providerSourcesToUninstall in sourcesToUninstall)
+            {
+                IReadOnlyList<UninstallResult> uninstallResults = await providerSourcesToUninstall.Key.UninstallAsync(providerSourcesToUninstall.Value, cancellationToken).ConfigureAwait(false);
+                foreach (UninstallResult uninstallResult in uninstallResults)
                 {
-                    var sourcesToUninstall = sources.Where(source => commandInput.ToUninstallList.Contains(source.Identifier, StringComparer.OrdinalIgnoreCase));
-                    if (!sourcesToUninstall.Any())
+                    if (uninstallResult.Success)
                     {
-                        continue;
+                        Reporter.Output.WriteLine($"Success: {uninstallResult.Source.DisplayName} was uninstalled.");
                     }
-                    sourceToUninstallFound = true;
-                    var uninstallResults = await provider.UninstallAsync(sourcesToUninstall, CancellationToken.None).ConfigureAwait(false);
-                    foreach (UninstallResult uninstallResult in uninstallResults)
+                    else
+                    {
+                        Reporter.Error.WriteLine(string.Format(LocalizableStrings.CouldntUninstall, uninstallResult.Source.DisplayName, uninstallResult.ErrorMessage));
+                        result = CreationResultStatus.CreateFailed;
+                    }
+                }
+            }
+            return result;
+        }
+
+        private async Task<(CreationResultStatus, Dictionary<IManagedTemplatesSourcesProvider, List<IManagedTemplatesSource>>)> DetermineSourcesToUninstall(INewCommandInput commandInput, CancellationToken cancellationToken)
+        {
```

Comment:
All errors in this method need to go via LocalizableStrings?

Reply:
Yes, I've created separate task for it #2880